core/trino-main/src/main/java/io/trino/execution/scheduler/TaskDescriptor.java

```diff
@@ -92,4 +99,17 @@ public String toString()
                 .add("nodeRequirements", nodeRequirements)
                 .toString();
     }
+
+    public long getRetainedSizeInBytes()
+    {
+        long result = retainedSizeInBytes;
+        if (result == 0) {
+            result = INSTANCE_SIZE
+                    + estimatedSizeOf(asMap(splits), PlanNodeId::getRetainedSizeInBytes, splits -> estimatedSizeOf(splits, Split::getRetainedSizeInBytes))
```

Comment:
would be nice to add support for `estimatedSizeOf` with MutliMap to airlift.

Reply:
`Multimap` is a Guava class. The `SizeOf` class is in the `slice` project that doesn't have a Guava dependency.